site/docs/sql/intro.md

```diff
@@ -1,351 +1,399 @@
+  yet support streaming sources like Kafka)
 
-In our topic the trades will be encoded as JSON messages:
+These are some of the features on our roadmap:
 
-```plain
-key: ABCD
-value: {
-    "ticker": "ABCD",
-    "price": 5.5,
-    "amount": 10
-}
-```
+- Joins with arbitrary external data sources
+- Windowed aggregation of streaming data
+- JDBC
 
-The `ticker` is repeated in both the key and the value. We'll ignore the
-key in the subsequent examples, it's only used by Kafka for
-partitioning.
+## CREATE EXTERNAL MAPPING
 
-### Creating the Mapping for Kafka Topic
+There is no native storage system in Jet SQL, instead it works with
+_external mappings_ to access various resources as if they were tables.
+This includes its own internal IMaps.
 
-To use a remote topic as a table in Jet, first create an `EXTERNAL
-MAPPING` for the topic. It maps the messages to a fixed list of columns
-with data types:
+This is how you can create a mapping to a Kafka topic `trades` with
+JSON messages:
 
 ```sql
-CREATE EXTERNAL MAPPING trades (
+sql> CREATE EXTERNAL MAPPING trades (
     ticker VARCHAR,
     price DECIMAL,
     amount BIGINT)
 TYPE Kafka
 OPTIONS (
     'valueFormat' = 'json',
     'bootstrap.servers' = '127.0.0.1:9092'
-    /* ... more configuration options for the Kafka consumer */
-)
+);
+OK
+sql> SHOW MAPPINGS;
++--------------------+
+|name                |
++--------------------+
+|tradeMap            |
++--------------------+
+1 row(s) selected
+sql>
 ```
 
-The `valueFormat` option specifies the serialization format for the
-value. For other possible values see the [Kafka
-Connector](kafka-connector) page. The options not handled by Jet, such
-as the `bootstrap.servers` above, are all passed directly to the Kafka
-consumer or producer.
+The [DDL](ddl) section has more details.
 
-To submit the above query, you can use the SQL CLI or the Java API (we
-also plan to support JDBC and non-Java clients in the future):
+## Query a CSV File
 
-#### Submitting the query on the CLI
-
-Before starting the CLI, make sure Hazelcast Jet server is running. If
-you don't have a running jet member, you can start a member using the
-following command:
+To try out the CSV file format, first activate the CSV connector:
 
 ```bash
-$JET_HOME/bin/jet-start
-...
+$ mv opt/hazelcast-jet-csv-4.4.jar lib/
+$
 ```
 
-After that, you can start the Jet SQL CLI by running this command:
+Restart the Jet server for this to take effect.
+
+Create a sample CSV file named `trades.csv`:
 
 ```bash
-$ $JET_HOME/bin/jet sql
-Connected to Hazelcast Jet 4.4-SNAPSHOT at [192.168.1.5]:5701 (+0 more)
+$ vi trades.csv
+
+id,name
+1,Jerry
+2,Greg
+3,Mary
+```
+
+Now you can write the SQL:
+
+```sql
+$ bin/jet sql
+Connected to Hazelcast Jet 4.4 at [192.168.5.13]:5701 (+0 more)
 Type 'help' for instructions
+sql> CREATE MAPPING csv_trades (id TINYINT, name VARCHAR)
+TYPE File
+OPTIONS ('format'='csv',
+    'path'='/path/to/hazelcast-jet-4.4', 'glob'='trades.csv');
```

Comment:
Probably better to use a different path
```suggestion
    'path'='/path/to/file', 'glob'='trades.csv');
```

Reply:
In the previous step the tutorial asked the user to create the file in this particular directory.